qiskit/pulse/schedule/pulse_schedule.py

```diff
@@ -91,34 +96,42 @@ def name(self) -> str:
     def channels(self) -> ChannelStore:
         return self._channel_store
 
-    def add(self,
-            commands: Union[PulseCommand, List[PulseCommand]],
-            channel: PulseChannel,
-            start_time: int):
-        """Add new pulse command(s) with channel and start time context.
+    def append(self, command: PulseCommand, channel: PulseChannel):
+        """Append a new pulse command on a channel at the timing
+        just after the last command finishes on the channel.
+
+        Args:
+            command (PulseCommand):
+            channel (PulseChannel):
+        """
+        try:
+            start_time = self.end_time_by(channel)  # TODO: need to add buffer?
+            self.add(TimedPulse(command, channel, start_time))
+        except ScheduleError as err:
+            logger.warning("Fail to append %s to %s", command, channel)
+            raise ScheduleError(err.message)
+
+    def insert(self, command: PulseCommand, channel: PulseChannel, start_time: int):
```

Comment:
`t0` should be the first argument.

Reply:
I like this order because "insert command at start_time" sounds natural for me. I guess you see "list.insert(index, to_be_inserted)", but I think time is different from index. For example, list cannot have several items at the same index but schedule can have several commands at the same timing. What do you think of it?

Reply:
Anyway, I've fixed this as you like so that we should be able to merge this first. Let's discuss this in the later PRs.